src/components/wizard/wizard.component.ts

```diff
@@ -347,6 +349,27 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
         return this.steps.toArray()[index];
     }
 
+    /**
+     * If a step in the wizard becomes invalid, all steps sequentially after
+     * it, should become unvisited and incomplete
+     */
+    protected setNextStepsUnvisited(currentStep: WizardStep): void {
```

Comment:
`WizardStep` interface enabled both Wizard and MarqueeWizard steps to be handled in the `WizardStepComponent` class.

Reply:
Hi Darren, I may have come up with a better way of doing this than requiring the WizardStep interface. It should be a fairly easy change. We can make the service use a generic instead eg:

```typescript
@Injectable()
export class WizardService<TWizardStep> {
    valid$ = new Subject<WizardValidEvent<TWizardStep>>();
}

export interface WizardValidEvent<TWizardStep> {
    step: TWizardStep;
    valid: boolean;
}
```

Then we can just define the type in the constructor, eg:

```typescript
constructor(wizardService: WizardService<MarqueeWizardStepComponent>,
        private _resizeService: ResizeService,
        private _elementRef: ElementRef<HTMLElement>)
```

This should keep the circular dependency out but mean we don't have to keep an additional interface maintained anytime either step component changes